crates/flipperzero/src/gui/mod.rs

```diff
@@ -1,3 +1,230 @@
 //! GUI service.
 
 pub mod canvas;
+pub mod icon;
+pub mod icon_animation;
+pub mod view;
+pub mod view_port;
+pub mod xbm;
+
+use crate::{
+    gui::{
+        canvas::CanvasView,
+        view_port::{ViewPort, ViewPortCallbacks},
+    },
+    input::InputEvent,
+};
+use core::{
+    ffi::c_char,
+    fmt::Debug,
+    ops::{Deref, DerefMut},
+};
+use flipperzero_sys::{
+    self as sys, furi::UnsafeRecord, Canvas as SysCanvas, Gui as SysGui, GuiLayer as SysGuiLayer,
+};
+
+/// System Gui wrapper.
+pub struct Gui {
+    raw: UnsafeRecord<SysGui>,
+}
+
+impl Gui {
+    /// Furi record corresponding to GUI.
+    pub const RECORD: *const c_char = sys::c_string!("gui");
+
+    pub fn new() -> Self {
```

Comment:
These new public APIs should all be documented. That would also help with review, knowing the intent behind each API (particularly the ones that aren't just a type-safe wrapper around a single SDK function).